pkg/agent/multicast/mcast_controller.go

```diff
@@ -596,3 +752,34 @@ func (c *Controller) GetAllPodsStats() map[*interfacestore.InterfaceConfig]*PodT
 	}
 	return statsMap
 }
+
+func memberExists(status *GroupMemberStatus, e *mcastGroupEvent) bool {
+	var exist bool
+	if e.iface.Type == interfacestore.ContainerInterface {
+		_, exist = status.localMembers[e.iface.InterfaceName]
+	} else if e.iface.Type == interfacestore.TunnelInterface {
+		exist = status.remoteMembers.Has(e.srcNode.String())
+	}
+	return exist
+}
+
+func addGroupMember(status *GroupMemberStatus, e *mcastGroupEvent) *GroupMemberStatus {
+	if e.iface.Type == interfacestore.ContainerInterface {
+		status.localMembers[e.iface.InterfaceName] = e.time
+		status.lastIGMPReport = e.time
+	} else {
+		status.remoteMembers = status.remoteMembers.Insert(e.srcNode.String())
+	}
+	return status
+}
+
+func deleteGroupMember(status *GroupMemberStatus, e *mcastGroupEvent) *GroupMemberStatus {
+	if e.iface.Type == interfacestore.ContainerInterface {
+		delete(status.localMembers, e.iface.InterfaceName)
+		klog.InfoS("Deleted local member from multicast group", "group", e.group.String(), "member", e.iface.InterfaceName)
```

Comment:
Does it really need to log every deletion event? It may be confusing to user that they see only "deletion" event but not "addition" event.